### PR TITLE
Allow resources to be updated.

### PR DESCRIPTION
**Pull request by tomason**
Added put method to http client and used it to update openshift resources.

### CODE REVIEW

File: src/main/java/com/openshift3/internal/client/DefaultClient.java [hunk 1]
```diff
@@ -137,6 +137,23 @@ private IHttpClient newIHttpClient(ISSLCertificateCallback sslCertCallback){
...
+		try {
+			final URL endpoint = new URLBuilder(getBaseURL(), getTypeMappings())
+				.resource(resource)
+				.addParmeter("namespace", resource.getNamespace())
+				.build();
+			String response = client.put(endpoint, IHttpClient.DEFAULT_READ_TIMEOUT, resource);
+			LOGGER.debug(response);
+			return factory.create(response);
+		} catch (HttpClientException e){
+			throw handleHttpClientException("Exception creating the resource", e);
```
Comment (jcantrill) [inline src/main/java/com/openshift3/internal/client/DefaultClient.java]: @tomason Update the error message(s) to reflect the operation (e.g. creating->update)

**Comment by jcantrill** [comment]
@tomason Please update the error messages and force push this PR and I will merge it tomorrow.  Thanks.

**Comment by tomason** [comment]
Copy-paste error that should be fixed now. Thank you for spotting this!

**Comment by jcantrill** [comment]
@tomason merged

**Pull request merged**